service/storage_proxy.cc

```diff
@@ -5254,9 +5254,8 @@ storage_proxy::handle_counter_mutation(const rpc::client_info& cinfo, rpc::opt_t
                 return remote().get_schema_for_write(schema_version, std::move(src_addr)).then([&mutations, fm = std::move(fm)] (schema_ptr s) mutable {
                     mutations.emplace_back(frozen_mutation_and_schema { std::move(fm), std::move(s) });
                 });
-            }).then([trace_state_ptr = std::move(trace_state_ptr), &mutations, cl, timeout] {
-                auto sp = get_local_shared_storage_proxy();
-                return sp->mutate_counters_on_leader(std::move(mutations), cl, timeout, std::move(trace_state_ptr), /* FIXME: rpc should also pass a permit down to callbacks */ empty_service_permit());
+            }).then([this, trace_state_ptr = std::move(trace_state_ptr), &mutations, cl, timeout] {
```

Comment:
The equivalent change would be to capture `shared_from_this()`, not `this`. If it's not required and `this` is enough, then a note in commit message would be good.